src/components/tooltip/Tooltip.tsx

```diff
@@ -290,25 +290,31 @@ class Tooltip extends React.Component<TooltipProps, State> {
             </>
         );
 
-        const tooltipWithStopBubble = (
+        const renderTooltip = () => (
+            <div className={classes} id={this.tooltipID} role="tooltip" aria-live="polite">
+                {tooltipInner}
+            </div>
+        );
+
+        const renderTooltipWithStopBubble = () => (
```

Comment:
Instead of separating the component like this and conditionally rendering, what about using the `stopBubble` prop in the `handleTooltipEvent`?

Reply:
Hi, yeah, I think that's another approach.
When this is originally started, I attached handleTooltipEvent to the role=tooltip div and in the handleTooltipEvent, I looked at stopBubble to decide whether I will stopBubble. However, lint complains that I can't attach event handler to role="tooltip", so I add a lint waiver. I got feedback from other parties that we should not waiver lint but instead create another div.

So a new div is created which changed layout, and once layout is changed, the priority becomes do not change existing behavior as much as I can, so it comes two different rendering path.

In the latest solution that I have two layout, for the legacy layout, I don't attach handleTooltipEvent, thus there is no need to check in the handleTooltipEvent that whether stopBubble is true or not.

Reply:
I am just wondering if this can be simplified. The major difference is that in this case, there is a wrapper div with event listeners attached. Can it always render this wrapper div and take this render path (remove the id, as i dont think its necessary?), and then deal with the bubbling in `handleTooltipEvent`

Reply:
hi, if we always render the wrapper div, we will change existing tooltip dom tree. This is bigger risk to take and potentially be a major version bump

Reply:
remove the new id